src/hooks/usePermalinkMember.ts

```diff
@@ -0,0 +1,116 @@
+ * In case of an event permalink it is the sender user Id of the event if that event is available.
+ * Otherwise returns null.
+ *
+ * @param type - pill type
+ * @param parseResult - permalink parse result
+ * @param event - permalink event, if available
+ * @returns permalink user Id. null if the Id cannot be determined.
+ */
+const determineUserId = (
+    type: PillType | null,
+    parseResult: PermalinkParts | null,
+    event: MatrixEvent | null,
+): string | null => {
+    if (type === null) return null;
+
+    if (parseResult?.userId) return parseResult.userId;
+
+    if (event && [PillType.EventInSameRoom, PillType.EventInOtherRoom].includes(type)) {
+        return event.getSender() ?? null;
+    }
+
+    return null;
+};
+
+/**
+ * Hook to get the permalink member
+ *
+ * @param type - Permalink type
+ * @param parseResult - Permalink parse result
+ * @param targetRoom - Permalink target room {@link ./usePermalinkTargetRoom.ts}
+ * @param event - Permalink event
+ * @returns The permalink member:
+ *          - The room member for a user mention
+ *          - The sender for a permalink to an event in the same room
+ *          - Null in other cases or the user cannot be loaded.
+ */
+export const usePermalinkMember = (
+    type: PillType | null,
+    parseResult: PermalinkParts | null,
+    targetRoom: Room | null,
+    event: MatrixEvent | null,
+): RoomMember | null => {
+    // User mentions and permalinks to events in the same room require to know the user.
+    // If it cannot be initially determined, it will be looked up later by a memo hook.
+    const shouldLookUpUser = type && [PillType.UserMention, PillType.EventInSameRoom].includes(type);
+    const userId = determineUserId(type, parseResult, event);
+    const userInRoom = shouldLookUpUser && userId && targetRoom ? targetRoom.getMember(userId) : null;
+    const [member, setMember] = useState<RoomMember | null>(userInRoom);
+
+    const doProfileLookup = useCallback((userId: string): void => {
+        MatrixClientPeg.get()
+            .getProfileInfo(userId)
+            .then((resp) => {
+                const newMember = new RoomMember("", userId);
+                newMember.name = resp.displayname || userId;
+                newMember.rawDisplayName = resp.displayname || userId;
+                newMember.getMxcAvatarUrl();
+                newMember.events.member = {
+                    getContent: () => {
+                        return { avatar_url: resp.avatar_url };
+                    },
+                    getDirectionalContent: function () {
+                        // eslint-disable-next-line
+                        return this.getContent();
+                    },
+                } as MatrixEvent;
+                setMember(newMember);
+            })
+            .catch((err) => {
+                logger.error("Could not retrieve profile data for " + userId + ":", err);
+            });
+    }, []);
+
+    useMemo(() => {
```

Comment:
It can be improved by using `useEffect` and moving out of the hook the `foundMember` which is computed from the `userId`. If `targetRoom?.getMember(userId)` has a cost, it can be useful to move it to his own `useMemo`

(pseudo code)

```ts
const foundMember = targetRoom?.getMember(userId) ?? null

useEffect(() => {
	if (!shouldLookUpUser || !userId || foundMember) {
            // nothing to do here
            return;
      }

     function doProfileLookup() {
        MatrixClientPeg.get()
            .getProfileInfo(userId)
            .then((resp) => {
                const newMember = new RoomMember("", userId);
                newMember.name = resp.displayname || userId;
                newMember.rawDisplayName = resp.displayname || userId;
                newMember.getMxcAvatarUrl();
                newMember.events.member = {
                    getContent: () => {
                        return { avatar_url: resp.avatar_url };
                    },
                    getDirectionalContent: function () {
                        // eslint-disable-next-line
                        return this.getContent();
                    },
                } as MatrixEvent;
                setMember(newMember);
            })
            .catch((err) => {
                logger.error("Could not retrieve profile data for " + userId + ":", err);
            });
     }
     
      doProfileLookup(userId)
}, [foundMember, doProfileLookup, shouldLookUpUser, targetRoom, userId])

return foundMember;
```